internal/pkg/cli/deploy/deploy.go

```diff
@@ -1210,8 +1214,13 @@ func validateAppVersionForAlias(appName string, appVersionGetter versionGetter)
 }
 
 func (d *backendSvcDeployer) validateALBRuntime() error {
-	if d.backendMft.RoutingRule.EmptyOrDisabled() || d.backendMft.RoutingRule.Alias.IsEmpty() || !d.env.HasImportedCerts() {
+	switch {
+	case d.backendMft.RoutingRule.EmptyOrDisabled():
 		return nil
+	case !d.backendMft.RoutingRule.Alias.IsEmpty() && !d.env.HasImportedCerts():
+		return fmt.Errorf(`setting "alias" requires an environment with imported certs`)
+	case d.backendMft.RoutingRule.HostedZone != nil && d.backendMft.RoutingRule.Alias.IsEmpty():
```

Comment:
Same question: perhaps  manifest `Validate` is a better home for them? What do ya think

Reply:
yeah, that looks like a better place - i'll move these over there 😊  same for the lbws one above.